store/postgres/src/deployment_store.rs

```diff
@@ -954,6 +947,33 @@ impl DeploymentStore {
         Ok(event)
     }
 
+    pub(crate) fn rewind(
+        &self,
+        site: Arc<Site>,
+        block_ptr_to: EthereumBlockPointer,
+    ) -> Result<StoreEvent, StoreError> {
+        let conn = self.get_conn()?;
+
+        self.rewind_with_conn(&conn, site, block_ptr_to)
+    }
+
+    pub(crate) fn revert_block_operations(
+        &self,
+        site: Arc<Site>,
+        block_ptr_to: EthereumBlockPointer,
+    ) -> Result<StoreEvent, StoreError> {
+        let conn = self.get_conn()?;
+        // Unwrap: If we are reverting then the block ptr is not `None`.
+        let block_ptr_from = Self::block_ptr_with_conn(&site.deployment, &conn)?.unwrap();
+
+        // Sanity check on block numbers
+        if block_ptr_from.number != block_ptr_to.number + 1 {
+            panic!("revert_block_operations must revert a single block only");
+        }
```

Comment:
Is this just here because we revert block by block usually? It's not here because reverting many blocks at once breaks things?

Reply:
It probably used to be that we had to go block-by-block; but for quite some time now, it's been safe to jump the block pointer back (assuming you know that you are jumping _to_ the correct block pointer, which this PR slyly puts on the user) All the reversion queries just look at things that happened at/after the given block.